### PR TITLE
SIP-94 Fixing underlying issue with settler on vSynth

### PR DESCRIPTION
**Pull request by justinjmoses**
1. Major: synth settlement wasn't being applied to the vSynth but rather the account who began the exchange, rendering the vSynth fairly useless.
2. Minor: vSynth to use proxy of synth to allow for future upgrades.
3. Incidental: optional tracking code for `exchangeWithVirtual`

### CODE REVIEW

**Review by jacko125 — DISMISSED** [review]
Approved.

File: contracts/Exchanger.sol [hunk 1]
```diff
@@ -580,7 +581,7 @@ contract Exchanger is Owned, MixinResolver, MixinSystemSettings, IExchanger {
 
         // persist the exchange information for the dest key
         appendExchange(
-            destinationAddress,
```
Comment (zyzek) [inline contracts/Exchanger.sol:583]: Do we not also need to do this when emitting the exchange event just above as well?

**Review by zyzek — CHANGES_REQUESTED** [review]
One small query. Also just to point out that we're going to need to be careful about settling all vSynths if we ever purge any Synths with outstanding settlements in future.

**Comment by jacko125** [comment]
> One small query. Also just to point out that we're going to need to be careful about settling all vSynths if we ever purge any Synths with outstanding settlements in future.

I think we should have a keeper settle and execute the vSynths to their final destinations as soon as possible. That was something discussed before so for the trader / user they wouldn't have to do anything manually to get the wBTC for example.

**Review by zyzek — APPROVED** [review]

**Review by jacko125 — APPROVED** [review]
Lgtm the changes

**Pull request merged**